ocs_sample_library_preview/Asset/Resolved/ResolvedStream.py

```diff
@@ -0,0 +1,57 @@
+import json
+
+from .ResolvedProperty import ResolvedProperty
+
+
+class ResolvedStream(object):
+    def __init__(self, name: str = None, properties: list[ResolvedProperty] = None):
+        self.Name = name
+        self.Properties = properties
+
+    @property
```

Comment:
some of the classes note which properties are required and which are optional. should this be done in all classes, such as this one?

Reply:
or is that only for classes that we expect a user to be regularly interacting with themselves, such as Asset

Reply:
We generally only have those comments on objects that have a create endpoint exposed, to document which properties are required. I didn't include them on objects that are only retrieved, since there's no 'required' properties for those objects.